cmake/traccc-compiler-options-cuda.cmake

```diff
@@ -31,6 +31,9 @@ traccc_add_flag( CMAKE_CUDA_FLAGS "--expt-relaxed-constexpr" )
 # build.
 traccc_add_flag( CMAKE_CUDA_FLAGS_DEBUG "-G" )
 
+#Allow the use of Nvidia extensions tools.
+traccc_add_flag( CMAKE_CUDA_FLAGS_DEBUG "-lnvToolsExt" )
```

Comment:
Does one need to link against the `libnvToolsExt.so` library to use `<nvtx3/nvToolsExt.h>`? :confused: I thought the `nvtxRangePush` / `nvtxRangePop` functions didn't need anything to be linked into the code. :confused:

We **could** use `CUDA::nvtx3`.

https://cmake.org/cmake/help/latest/module/FindCUDAToolkit.html#nvtx3

Just privately linking `traccc_performance` against that I mean. But:
  - That will force us to use CMake 3.25 in our builds;
  - That still only talks about a "header only" library. So I still don't understand why this linking is needed. :confused:

Reply:
Looks like its working perfectly fine without it. I think I forgot to remove the link after inserting:
`find_package(CUDAToolkit)
if (CUDAToolkit_FOUND)
   target_link_libraries( traccc_performance PRIVATE CUDA::cudart )
   target_compile_definitions( traccc_performance PRIVATE TRACCC_HAVE_NVTX )
endif()`
in performance/CMakeLists.txt 😑